docs/spec/v2beta1/helmreleases.md

```diff
@@ -805,6 +823,98 @@ spec:
         memory: 64Mi
 ```
 
+## Remote Clusters / Cluster-API
+
+If the `spec.kubeConfig` field is set, Helm actions will run against the default cluster specified
+in that KubeConfig instead of the local cluster that is responsible for the reconciliation of the
+HelmRelease.
+
+The secret defined in the `spec.kubeConfig.secretRef` should exist in the same namespace as the
+HelmRelease. On every reconciliation, the KubeConfig bytes will be loaded from the `values` key
+of the secret's data, and the secret can thus be regularly updated if cluster-access-tokens have
+to rotate due to expiration. 
+
+The Helm storage is stored on the remote cluster in a namespace that equals to the namespace of
+the HelmRelease, the release itself is made in either this namespace, or the configured
+`spec.targetNamespace`. In any case, both are expected to exist.
```

Comment:
I discussed this with Scott awhile ago, and there does not seem to be such a proposal.

I do however think it would be smart to discuss this upstream (or draft and submit a [HIP](https://github.com/helm/community/blob/master/hips/hip-0001.md)), especially because the way we currently achieve it may break in the future (as we basically trick the client by defaulting to another namespace than the namespace we use for storage).